### PR TITLE
Register multiple Redis client instances with Typescript

### LINKED ISSUE #102: How to register multiple Redis client instances with Typescript
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure it has not already been reported

### Fastify version

3.19.2

### Plugin version

4.3.1

### Node.js version

v14.16.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

11.4

### Description

I'm trying to register multiple Redis client instances as follow:

```typescript
export default fp<FastifyRedisPlugin>(async (fastify) => {
  await fastify
    .register(fastifyRedis, {
      host: fastify.config.REDIS_HOST,
      port: fastify.config.REDIS_PORT,
      password: fastify.config.REDIS_PASSWORD,
      namespace: 'hello',
    })
    .register(fastifyRedis, {
      host: fastify.config.REDIS_HOST,
      port: fastify.config.REDIS_PORT,
      password: fastify.config.REDIS_PASSWORD,
      namespace: 'world',
    });
}
```

I would expect that, as explained in the docs, I would be able to do something like that:

````typescript
// Here we will use the `hello` named instance
fastify.get('/hello', (req, reply) => {
  const { redis } = fastify

  redis.hello.get(req.query.key, (err, val) => {
    reply.send(err || val)
  })
})

fastify.post('/hello', (req, reply) => {
  const { redis } = fastify

  redis['hello'].set(req.body.key, req.body.value, (err) => {
    reply.send(err || { status: 'ok' })
  })
})

// Here we will use the `world` named instance
fastify.get('/world', (req, reply) => {
  const { redis } = fastify

  redis['world'].get(req.query.key, (err, val) => {
    reply.send(err || val)
  })
})
````

But whenever I try to access `redis['hello']` TypeScript complains about that with the following error:
````
Element implicitly has an 'any' type because expression of type '"hello"' can't be used to index type 'Redis'.
  Property 'hello' does not exist on type 'Redis'.
````

I thought that maybe I would have been able to fix this issue by myself doing that, albeit this is not documented anywhere:

````typescript

import type { Redis } from 'ioredis'

declare module 'fastify' {
  interface FastifyInstance {
    redis: {
      hello: Redis;
      world: Redis;
    };
  }
}
````

But this time TypeScript greets me with this error:

```
Subsequent property declarations must have the same type.  Property 'redis' must be of type 'Redis', but here has type '{ test: Redis; test2: Redis; }'.
```

So this seems a dead end to me.

Can someone shed a light on this for me?

Thanks in advance.

### Steps to Reproduce

Just follow the [Registering multiple Redis client instances](https://github.com/fastify/fastify-redis#registering-multiple-redis-client-instances) step in Readme.MD with a project bootstrapped with [fastify-cli](https://github.com/fastify/fastify-cli) and `TypeScript`.

### Expected Behavior

I expect that one can register multiple Redis instances and correctly access them, even when using `TypeScript`.

### PR DESCRIPTION
**Pull request by CloudPower97**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

#### Description

Closes #102

### CODE REVIEW

**Comment by CloudPower97** [comment]
Hi, @Eomm I'm not sure why but all of the tests are failing no matter what.

I'm using the official [`Redis` Docker image](https://hub.docker.com/_/redis), exposing port `6379` to the host.

But I receive the following error:

````javascript
  node_modules/avvio/plugin.js                                                                       
  121 |       debug('timed out', name)                                                               
  122 |       timer = null                                                                           
> 123 |       const err = new Error(`${CODE_PLUGIN_TIMEOUT}: plugin did not start in time: ${name}. You may have forgotten to call 'done' function or to resolve a Promise`)
      | ------------------^                                                                          
  124 |       err.code = CODE_PLUGIN_TIMEOUT                                                         
  125 |       err.fn = func                                                                          
  126 |       done(err)   
````

Any ideas?

**Review by climba03003 — CHANGES_REQUESTED** [review]
Please remove the unnecessary linting change and please run ``npm run lint -- --fix` before commit.

File: test/test.js [hunk 1]
```diff
@@ -449,14 +463,37 @@ test('Should throw authentication error when trying to connect on a valid host w
...
-        t.ok(err)
-      })
+    fastify.ready((err) => {
+      t.ok(err)
     })
+  })
+})
+
+test('Registering multiple Redis client instances', (t) => {
+  t.plan(2)
```
Comment (Eomm) [inline test/test.js:478]:
```suggestion
  t.plan(3)
```

**Review by Eomm — CHANGES_REQUESTED** [review]
I think it is missing a test here
https://github.com/fastify/fastify-redis/blob/master/test/types/index.ts

The CI is failing due the a test plan (check my suggestion)

**Review by climba03003 — COMMENTED** [review]
Can you add the typescript test in this file?
https://github.com/fastify/fastify-redis/blob/master/test/types/index.ts

**Pull request closed without merging**